stglib/aqd/aqdutils.py

```diff
@@ -1420,3 +1435,30 @@ def apply_wave_coord_output(ds, T, T_orig):
         ds = ds.drop(["VEL1", "VEL2", "VEL3"])
 
     return ds
+
+
+def fill_agc(ds):
+    """
+    Fill velocity data with AGC threshold.
+    Beam AGC variables (AGC1_1221, AGC2_1222, and AGC3_1223) are used to fill corresponding beam velocities (vel1_1277, vel2_1278, vel3_1279).
+    Average AGC (AGC_1202) is used to fill transformed eastward, northward, and upward velocities (u_1205, v_1206, w_1204).
+    """
+    if "agc_threshold" in ds.attrs:
+        Ptxt = str(ds.attrs["agc_threshold"])
+
+        # fill transformed u,v,w velocities by averaged AGC (AGC_1202)
+        uvw = ["u_1205", "v_1206", "w_1204"]
+
+        for var in uvw:
+            ds[var] = ds[var].where(ds["AGC_1202"] > ds.attrs["agc_threshold"])
+            notetxt = "Data filled using AGC_1202 threshold of %s counts. " % (
```

Comment:
For new code, suggest using f strings instead of the old % way of inserting variables.